### ISSUE TITLE
Sequence Tween.OnComplete early call when Time.timeScale = 15

### ISSUE THREAD

**Issue by ctxdegithub** [KyryloKuzyk/PrimeTween#21]
Sorry for my poor english.
the code is that
```cs
Time.timeScale = 15;
var myTrans = transform;
myTrans.localScale = new Vector3(0.2f, 0.2f, 0.2f);
Sequence
    .Create(Tween.Scale(myTrans, 1.2f, 0.2f, Ease.OutSine).OnComplete(() => { Debug.Log("Tween1 " + myTrans.localScale); }))
    .Chain(Tween.Scale(myTrans, 1, 0.1f, Ease.OutBack).OnComplete(() => { Debug.Log("Tween2 " + myTrans.localScale); })
    );
```
I set timeScale to 15, , but Tween2 prints out before Tween1, which is the reverse order.

**Comment by KyryloKuzyk** [comment]
Hmm, I just tested it myself in a stress test, but I can't replicate the issue. Do you have a self-contained reproducible example, like this one?
```csharp
using System.Collections;
using PrimeTween;
using UnityEngine;

public class PrimeTweenExample : MonoBehaviour {
    [SerializeField] float timeScale = 15f;
    
    IEnumerator Start() {
        for (int i = 0; i < 100; i++) {
            print("-------------------------");
            Time.timeScale = timeScale;
            var myTrans = transform;
            myTrans.localScale = new Vector3(0.2f, 0.2f, 0.2f);
            yield return Sequence.Create(Tween.Scale(myTrans, 1.2f, 0.2f, Ease.OutSine).OnComplete(() => Debug.Log($"Frame {Time.frameCount}, Tween1 " + myTrans.localScale)))
                .Chain(Tween.Scale(myTrans, 1, 0.1f, Ease.OutBack).OnComplete(() => Debug.Log($"Frame {Time.frameCount}, Tween2 " + myTrans.localScale)));
            yield return null;
        }
    }
}
```

**Comment by KyryloKuzyk** [comment]
Can you please update to version 1.0.17 and tell me if your issue still exists?

**Comment by ctxdegithub** [comment]
Hi, I apologize for the late response. I have replicated the bug.
I create an empty GameObject and Add TweenTest component to it.
When using Sequence, I first created Tween2 before creating Tween1, and then added it to the Sequence. With the following code, after running it, there is a chance that this problem occurs when pressing the space bar multiple times.
```cs
using PrimeTween;
using UnityEngine;

public class TweenTest : MonoBehaviour
{
    private int _nextTweenId;
    void Start()
    {
        Time.timeScale = 10;
        Time.maximumDeltaTime = 0.1f;
    }

    void Update()
    {
        if (Input.GetKeyDown(KeyCode.Space))
        {
            ++_nextTweenId;
            var myTrans = new GameObject("Tween" + _nextTweenId).transform;
            myTrans.localScale = new Vector3(0.2f, 0.2f, 0.2f);
            var scale2 = Tween.Scale(myTrans, 1, 0.1f).OnComplete(() =>
            {
                Debug.Log("Tween2 " + myTrans.localScale);
                myTrans.gameObject.name += ", 2: " + myTrans.localScale;
            });
            var scale1 = Tween.Scale(myTrans, 1.2f, 0.2f).OnComplete(() =>
            {
                Debug.Log("Tween1 " + myTrans.localScale);
                myTrans.gameObject.name += ", 1: " + myTrans.localScale;
            });
            Sequence.Create(scale1).Chain(scale2);
        }
    }
}
```
![image](https://github.com/KyryloKuzyk/PrimeTween/assets/8472028/7173be2c-d40e-429f-97b2-fbca2e6dbcf6)
![image](https://github.com/KyryloKuzyk/PrimeTween/assets/8472028/f649f5d4-7d5c-4b27-963b-89a9c058b90c)
I think it's the creation order that led to the incorrect update sequence of Tweens by PrimeTweenManager.update, which resulted in Tween2's callback being completed first.
@KyryloKuzyk

**Comment by ctxdegithub** [comment]
The PrimeTween version is 1.0.16（our game project） and 1.1.0.

**Comment by KyryloKuzyk** [comment]
@ctxdegithub Huge thanks for the great reproducible test!

I was able to reproduce the issue on my end with version 1.0.16. Unfortunately, this version was corrupted because I accidentally released it from the work-in-progress repository. I'm very sorry for that.

The issue was already fixed in version 1.0.17. And the latest version 1.1.1 also doesn't have this issue. **Could you please confirm that?**

I edited your test slightly, so it's easier to reproduce the issue:
```csharp
using PrimeTween;
using UnityEngine;
using UnityEngine.Assertions;

public class TweenTest : MonoBehaviour {
    void Start() {
        Application.targetFrameRate = 30; // the bug reproduces more consistently with lower fps
        Time.timeScale = 10;
        Time.maximumDeltaTime = 0.1f;
    }

    void Update() {
        int order = 0;
        var tween2 = Tween.Delay(1, 0.1f).OnComplete(() => {
            Assert.AreEqual(1, order); // this assertion fails in version 1.0.16 because tween2 finishes before tween1 
            order++;
        });
        var tween1 = Tween.Delay(1.2f, 0.2f).OnComplete(() => {
            Assert.AreEqual(0, order);
            order++;
        });
        Sequence.Create(tween1).Chain(tween2);
    }
}
```

**Comment by ctxdegithub** [comment]
Thank you for your reply. This issue no longer exists in version 1.1.1.
